toolbox/_common.py

```diff
@@ -0,0 +1,98 @@
+import itertools
+import subprocess
+import os
+import time
+import sys
+from pathlib import Path
+
+
+top_dir = Path(__file__).resolve().parent.parent
+
+
+class PlaybookRun:
+    """
+    Playbook runner
+
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    If you're seeing this text, put the --help flag earlier in your list
+    of command-line arguments, this is a limitation of the CLI parsing library
+    used by the toolbox.
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+    """
+    def __init__(self, playbook_name, opts: dict = dict()):
+        self.playbook_name = playbook_name
+        self.opts = opts
+
+    def __str__(self):
+        return ""
+
+    def _run(self):
+        run_ansible_playbook(self.playbook_name, self.opts)
+
+
+def flatten(lst):
+    return itertools.chain(*lst)
+
+
+def run_ansible_playbook(playbook_name, opts: dict = dict()):
```

Comment:
@omertuc I think you shouldn't use `dict()` as default value, but `=None` and `if opts is None: opts = dict()`:
```
def hello(d=dict()):
   global i
   i += 1
   d[i] = i
   print(d)

i = 0
hello()
hello()
hello()
```
gives:
```
{1: 1}
{1: 1, 2: 2}
{1: 1, 2: 2, 3: 3}
```
see the first section of this [page](https://docs.python-guide.org/writing/gotchas/) (and let me know if I missed something ^^)

I guess the issue is transparent if `run_ansible_playbook` is called only once, but still ;-)

Reply:
Yeah I'm aware of this quirk of Python. I thought it only applied when using `{}`, that's why I used `dict()`, for some reason I believed it would prevent that.

ack

Reply:
I had to triple check to be sure it applied here, and defining functions from the interpreter prompt isn't the most efficient way to write Python :/

that's a strange default behavior, I wonder where it came from ...!